service/history/api/respondactivitytaskcompleted/api.go

```diff
@@ -97,7 +97,6 @@ func Invoke(
 			}
 
 			if !isRunning ||
-				ai.StartedEventId == common.EmptyEventID ||
```

Comment:
You don't want to always ignore this, only disable this check if the token is a token generated by workflow_handler when handling a `RespondActivityTaskCompletedById`.

Reply:
No this is `StartedEventId` on the activity info struct not on the token.